lib/src/chart/line_chart/line_chart_data.dart

```diff
@@ -697,7 +697,7 @@ typedef GetDotColorCallback = Color Function(FlSpot, double, LineChartBarData);
 /// If there is one color in [LineChartBarData.colors], it returns that color,
 /// otherwise it returns the color along the gradient colors based on the [xPercentage].
 Color _defaultGetDotColor(FlSpot _, double xPercentage, LineChartBarData bar) {
-  if (bar.colors.isEmpty || bar.colors.isEmpty) {
+  if (bar.colors.isEmpty) {
     return Colors.green;
```

Comment:
And I think here is better to throw an exception (I know it is out of this PR's scope)

Reply:
done